src/repository/ParticipantRepository.spec.ts

```diff
@@ -0,0 +1,89 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { MongoMemoryServer } from 'mongodb-memory-server';
+import DatabaseClient from "../database";
+import ParticipantRepository from './ParticipantRepository';
+import { Participant } from '../types/participant';
+
+const mongod = new MongoMemoryServer();
+
+async function asyncForEach(array: Array<string>, callback): Promise<void> {
+  const results = [];
+  for (let index = 0; index < array.length; index += 1) {
+    results.push(callback(array[index], index, array));
+  }
+  await Promise.all(results);
+}
+
+describe("Participant Repository", () => {
+  let client: DatabaseClient;
+
+  beforeAll(async () => {
+    const url = await mongod.getUri();
+    const database = await mongod.getDbName();
+
+    client = new DatabaseClient({
+      url,
+      database,
+    });
+    await client.connect();
+
+  });
+
+  beforeEach(async () => {
+    await client.getCollection("participants").deleteMany({}) // Clean the collection in the test database
+  })
+
+  afterAll(async () => {
+    await client.close(true);
+    await mongod.stop();
+  });
+  
+  it("should init the repository",async () => {
+    ParticipantRepository.init(client)
+    expect(ParticipantRepository.participantCollection).not.toBeUndefined()
+  })
+
+  it('should increment participant points by 1', async () => {
+    await ParticipantRepository.participantCollection.insertOne({name:"LucasMontano", points: 1})
+
+    await ParticipantRepository.incrementParticipantPoints("LucasMontano")
+
+    const LucasMontanospoints = (await ParticipantRepository.participantCollection.findOne({name: "LucasMontano"})).points
+    expect(LucasMontanospoints).toBe(2);
+  });
+
+  it('should fetch the top 10', async () => {
+    const top10 = [
```

Comment:
a library to return random names is a great option for this case, such as faker and factory-girl